PowerballApi.Tests/PowerballRepositoryTests.cs

```diff
@@ -1,140 +1,271 @@
+                PowerPlay = null,
+                WinNumbers = new[] { 1, 2, 3, 4, 5, 6 }
+            };
+            _cacher.Get(Arg.Any<string>()).Returns(new List<PowerballSet> { drawing });
+
+            var result = _sut.Get(requestDate);
+
+            Assert.IsNull(result);
+        }
+
+        [TestMethod]
+        public void GetById_WhenMultipleMatchingDrawings_ThrowException()
+        {
+            const string date = "12/12/2000";
+            var drawing = new PowerballSet
+            {
+                Date = DateTime.Parse(date),
+                PowerPlay = null,
+                WinNumbers = new[] { 1, 2, 3, 4, 5, 6 }
+            };
+            _cacher.Get(Arg.Any<string>()).Returns(new List<PowerballSet>
+            {
+                drawing,
+                drawing
+            });
+
+            try
+            {
+                _sut.Get(date);
+            }
+            catch (Exception ex)
+            {
+                Assert.IsInstanceOfType(ex, typeof(Exception));
+            }
+        }
+
+        [TestMethod]
+        public void GetByRange_InvalidAfterDate_ThrowException()
+        {
+            const string dateAfter = "12/50/2011";
+            const string dateBefore = "12/20/2013";
+
+            var drawing = new PowerballSet
+            {
+                Date = DateTime.Parse(dateBefore),
+                PowerPlay = null,
+                WinNumbers = new[] { 1, 2, 3, 4, 5, 6 }
+            };
+            _cacher.Get(Arg.Any<string>()).Returns(new List<PowerballSet>
+
+            {
+                drawing
+            });
+
+            try
+            {
+                _sut.GetByRange(dateAfter, dateBefore);
+            }
+            catch (Exception ex)
+            {
+
+                Assert.IsInstanceOfType(ex, typeof(Exception));
+            }
+        }
+
+        [TestMethod]
+        public void GetByRange_InvalidBeforeDate_ThrowsException()
+        {
+            const string dateAfter = "01/15/2015";
+            const string dateBefore = "01/44/2016";
+
+            var drawing = new PowerballSet
+            {
+                Date = DateTime.Parse(dateAfter),
+                PowerPlay = null,
+                WinNumbers = new[] { 1, 2, 3, 4, 5, 6 }
+            };
+            _cacher.Get(Arg.Any<string>()).Returns(new List<PowerballSet>
+
+            {
+                drawing
+            });
+
+            try
+            {
+                _sut.GetByRange(dateAfter, dateBefore);
+            }
+            catch (Exception ex)
+            {
+
+                Assert.IsInstanceOfType(ex, typeof(Exception));
+            }
+        }
+
+        [TestMethod]
+        public void GetByRange_ValidDates_ReturnsDrawingsInRange()
+        {
+            const string dateAfter = "01/01/2017";
+            const string dateBefore = "01/31/2017";
+
+            var drawingList = new List<PowerballSet>
+            {
+                new PowerballSet
```

Comment:
In order to sufficiently test a drawing within range is returned, you only need one item in the list, two at the most.